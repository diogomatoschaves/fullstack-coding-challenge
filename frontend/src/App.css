.app {
  width: 100%;
  margin-top: 140px;
  /*text-align: center;*/
  /*min-height: 500px*/
}

.title {
  font-size: 2.0em;
  position: absolute;
  left: 10px;
  top: 10px;
  background-color: #3351c6;
  /*background-color: #133480;*/
  /*text-decoration: underline;*/
}

.flex-column {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.flex-row {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.message-header {
  font-size: 1.2em;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 300px;
  overflow: hidden;
}

.message-to-user {
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  font-weight: 500;
  position: fixed;
  width: 50%;
  left: 25%;
  height: 60px;
}