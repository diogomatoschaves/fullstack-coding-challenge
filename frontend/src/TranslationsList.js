/**
 * Created by diogomatoschaves on 24/11/2018.
 */


import React, { Component } from 'react'
import { ListGroup, ListGroupItem, Button, ButtonGroup, Well } from 'react-bootstrap'


class TranslationsList extends Component {

  static defaultProps = {
    styleOptions: {
      requesting: 'warning',
      pending: 'info',
      completed: 'success',
      failed: 'danger'
    }
  }

  deleteJob = ({ id }) => {
  
    const headers = new Headers()
    
    const { deleteTranslation } = this.props
    
    const url = `${process.env.NODE_ENV === 'development' ? process.env.REACT_APP_API_URL_DEVELOPMENT 
      : process.env.REACT_APP_API_URL_PRODUCTION}/delete_job?id=${id}`;
    
    deleteTranslation({ id })
    
    fetch(url, {
      method: 'get',
      headers,
    })
      .then(res => res.json())
      .then(response => {
        if (response.status === 'deleted') {
          
        }
      })
  }

  getFormattedDate = (timeStamp) => {

    let difference = Date.now() - timeStamp;
    const daysDifference = Math.floor(difference/1000/60/60/24);
    difference -= daysDifference*1000*60*60*24;

    const hoursDifference = Math.floor(difference/1000/60/60);
    difference -= hoursDifference*1000*60*60;

    const minutesDifference = Math.floor(difference/1000/60);
    difference -= minutesDifference*1000*60;

    const secondsDifference = Math.floor(difference/1000);

    return daysDifference >= 1 ? `${daysDifference} ${daysDifference === 1 ? 'day' : 'days'} ago` :
        hoursDifference >= 1 ? `${hoursDifference} ${hoursDifference === 1 ? 'hour' : 'hours'} ago` :
            minutesDifference >= 1 ? `${minutesDifference} ${minutesDifference === 1 ? 'minute' : 'minutes'} ago` :
                `${secondsDifference} ${secondsDifference === 1 ? 'second' : 'seconds'} ago`

  };
  
  render() {

    const { translations, styleOptions, expandTranslation, checkStatus } = this.props

    // const sortProp = optionsSort.filter((option) => option.value === valueSort)[0].key;

    const orderedTranslations = translations ? Object.keys(translations).map(key => translations[key]).sort((a, b) => {
      return b.originalText.length - a.originalText.length
    }) : []
    
    return (
      <div className="flex-row" style={{width: '80%', maxWidth: '850px', justifyContent: 'space-around'}}>
        <ListGroup style={{width: '100%'}}>
          {orderedTranslations && orderedTranslations.map(translation => {

            return (
              <div key={translation.id} className="flex-row translation-item-wrapper" style={{width: '100%', justifyContent: 'space-between'}}>
                <ListGroupItem
                  bsStyle={styleOptions[translation.status]}
                  style={{cursor: 'pointer', padding: 0, width: '77%', border: translation.expanded && '2px solid rgb(180, 180, 180)'}}
                  className="flex-column translation-item"
                >
                  <div onClick={() => expandTranslation(translation.id)} className="flex-column" style={{width: '100%', padding: '20px'}}>
                    <div className='flex-row' style={{width: '100%', justifyContent: 'space-between'}}>
                      <div style={{width: '40%'}} className="message-header">{translation.originalText}</div>
                      <div style={{width: '25%', textAlign: 'center'}}><strong>time: </strong>{this.getFormattedDate(translation.timeStamp)}</div>
                      <div style={{width: '30%', textAlign: 'right'}}><strong>status: </strong>{translation.status}</div>
                    </div>
                    {translation.expanded && (
                      <div style={{width: '100%', marginTop: '15px'}}>
                        <strong>Original Text:</strong>
                        <Well style={{width: '100%'}}>
                          {translation.originalText}
                        </Well>
                        {translation.translatedText && (
                          <div style={{width: '100%', marginTop: '15px'}}>
                            <strong>Translated Text:</strong>
                            <Well style={{width: '100%'}}>
                              {translation.translatedText}
                            </Well>
                          </div>
                        )}
                      </div>
                    )}
                  </div>
                </ListGroupItem>
                <div style={{width: '20%', textAlign: 'center', alignSelf: 'flex-start', paddingTop: '20px'}}>
                  <ButtonGroup style={{whiteSpace: 'nowrap'}}>
                    <Button
                      onClick={() => checkStatus([{id: translation.id, uid: translation.uid}])}
                      disabled={translation.status !== 'pending' || translation.disabled}
                      bsStyle="info"
                      // style={{maxWidth: '100px'}}
                    >
                      {translation.disabled ? 'Checking..' : 'Update'}
                    </Button>
                    <Button 
                      bsStyle="danger"
                      onClick={() => this.deleteJob({ id: translation.id })}
                    >
                      Delete
                    </Button>
                  </ButtonGroup>
                </div>
              </div>
            )
          })}
        </ListGroup>
      </div>
    )
  }
}

export default TranslationsList

