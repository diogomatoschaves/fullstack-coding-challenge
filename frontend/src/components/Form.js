/**
 * Created by diogomatoschaves on 24/11/2018.
 */

import React, { Component } from 'react'
import { FormGroup, ControlLabel, FormControl, Button, ListGroupItem} from 'react-bootstrap'


class Form extends Component {

  static defaultProps = {
    sourceLang: 'en',
    targetLang: 'es'
  }

  state = {
    message: '',
    text: ''
  }

  getResult = ({ jobId, id, text }) => {

    const { addUid, addTranslation, updateTranslation, checkStatus, sourceLang, targetLang } = this.props
    let i = 0

    addTranslation({ id, jobId, originalText: text, sourceLang, targetLang, status: 'requesting' })

    this.setState({
      showMessage: true,
      message: 'Your request for translation was sent!',
      bsStyle: 'success'
    })
    setTimeout(() => {
      this.setState({ showMessage: false })
    }, 3000)

    const url = `${process.env.NODE_ENV === 'development' ? process.env.REACT_APP_API_URL_DEVELOPMENT 
      : process.env.REACT_APP_API_URL_PRODUCTION}/check_confirmation?jobId=${jobId}&id=${id}`;

    const headers = new Headers()

    const timeoutCallback = () => {
      i++
      fetch(url, {
        method: 'get',
        headers,
      })
        .then(res => res.json())
        .then(response => {
          if (response.uid) {
            addUid(id, response.uid)
            setTimeout(() => {
              checkStatus([{ id, uid: response.uid }])
            }, 3000)
          } else {
            if (response.status && response.status === 'failed') {
              updateTranslation(id, {
                status: 'failed',
              })
            } else {
              const { translations } = this.props
              if (Object.keys(translations).includes(id)) setTimeout(timeoutCallback, (7 - i) >= 2 ? (7 - i)*1000 : 2000)
            }
          }
        })
    }
    setTimeout(timeoutCallback, 7000)
  }

  buttonSubmit = () => {

    const { text } = this.state
    const { sourceLang, targetLang } = this.props

    this.setState({ text: '' })

    if (!text) return

    const headers = new Headers()
    
    const url = `${process.env.NODE_ENV === 'development' ? process.env.REACT_APP_API_URL_DEVELOPMENT 
      : process.env.REACT_APP_API_URL_PRODUCTION}/new_translation`;

    const body =  JSON.stringify({ text, sourceLang, targetLang })
    
    fetch(url, {
      method: 'post',
      headers,
      body
    })
      .then(res => res.json())
      .then(response => this.getResult({ jobId: response.job_id, id: response.translation_job, text }))
      .catch(error => {
        this.setState({
          showMessage: true,
          message: 'There was an error processing your request. Make sure you are connected to the internet.',
          bsStyle: 'danger'
        })
        setTimeout(() => {
          this.setState({ showMessage: false })
        }, 3000)
      })
  }

  handleChange = (e) => {
    this.setState({ text: e.target.value });
  }
  
  render() {

    const { text, message, showMessage, bsStyle } = this.state
    
    return (
      <div className="flex-row" style={{width: '80%', maxWidth: '800px', justifyContent: 'space-around'}}>
        <FormGroup style={{width: '77%'}}>
          <ControlLabel>Text to be translated</ControlLabel>
          <FormControl
            value={text}
            onChange={this.handleChange}
            componentClass="textarea"
            placeholder="Insert your text here"
            style={{height: '150px'}}
          />
        </FormGroup>
        <div className="flex-row" style={{width: '20%'}}>
          <Button disabled={!text} bsStyle="primary" bsSize="large" onClick={this.buttonSubmit}>
            Request
          </Button>
        </div>
        <ListGroupItem
          className='message-to-user flex-column'
          bsSize="large"
          bsStyle={bsStyle}
          style={{top: showMessage ? '60px' : '-200px'}}
        >
          {message}
        </ListGroupItem>
      </div>
    )
  }
}

export default Form
