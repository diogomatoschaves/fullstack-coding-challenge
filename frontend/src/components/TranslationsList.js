/**
 * Created by diogomatoschaves on 24/11/2018.
 */


import React, { Component } from 'react'
import { ListGroup, ListGroupItem, Button, ButtonGroup, Well } from 'react-bootstrap'


class TranslationsList extends Component {

  static defaultProps = {
    styleOptions: {
      requesting: 'warning',
      pending: 'info',
      completed: 'success',
      failed: 'danger'
    }
  }

  deleteJob = ({ id }) => {
  
    const headers = new Headers()
    
    const { deleteTranslation } = this.props
    
    const url = `${process.env.NODE_ENV === 'development' ? process.env.REACT_APP_API_URL_DEVELOPMENT 
      : process.env.REACT_APP_API_URL_PRODUCTION}/delete_job?id=${id}`;
    
    deleteTranslation({ id })
    
    fetch(url, {
      method: 'get',
      headers,
    })
      .then(res => res.json())
      .then(response => {
        if (response.status === 'deleted') {
          
        }
      })
  }
  
  render() {

    const { translations, styleOptions, expandTranslation, checkStatus } = this.props
    
    return (
      <div className="flex-row" style={{width: '80%', maxWidth: '850px', justifyContent: 'space-around'}}>
        <ListGroup style={{width: '100%'}}>
          {translations && Object.keys(translations).map(translation => {

            const data = translations[translation]

            return (
              <div className="flex-row" style={{width: '100%', justifyContent: 'space-between'}}>
                <ListGroupItem
                  key={translation}
                  bsStyle={styleOptions[data.status]}
                  style={{cursor: 'pointer', padding: 0, width: '77%'}}
                  className="flex-column"
                >
                  <div onClick={() => expandTranslation(data.id)} className="flex-column" style={{width: '100%', padding: '20px'}}>
                    <div className='flex-row' style={{width: '100%', justifyContent: 'space-between'}}>
                      <div style={{width: '60%'}} className="message-header">{data.originalText}</div>
                      <div style={{width: '30%'}}><strong>status: </strong>{data.status}</div>
                    </div>
                    {data.expanded && (
                      <div style={{width: '100%', marginTop: '15px'}}>
                        <strong>Original Text:</strong>
                        <Well style={{width: '100%'}}>
                          {data.originalText}
                        </Well>
                        {data.translatedText && (
                          <div style={{width: '100%', marginTop: '15px'}}>
                            <strong>Translated Text:</strong>
                            <Well style={{width: '100%'}}>
                              {data.translatedText}
                            </Well>
                          </div>
                        )}
                      </div>
                    )}
                  </div>
                </ListGroupItem>
                <div style={{width: '20%'}}>
                  <ButtonGroup style={{whiteSpace: 'nowrap'}}>
                    <Button
                      onClick={() => checkStatus([{id: data.id, uid: data.uid}])}
                      disabled={data.status !== 'pending' || data.disabled}
                      bsStyle="info"
                      // style={{maxWidth: '100px'}}
                    >
                      {data.disabled ? 'Checking..' : 'Update'}
                    </Button>
                    <Button 
                      bsStyle="danger"
                      onClick={() => this.deleteJob({ id: data.id })}
                    >
                      Delete
                    </Button>
                  </ButtonGroup>
                </div>
              </div>
            )
          })}
        </ListGroup>
      </div>
    )
  }
}

export default TranslationsList

