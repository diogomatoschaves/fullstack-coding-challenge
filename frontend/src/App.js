import React, { Component } from 'react';
import './App.css';
import { Label } from 'react-bootstrap'
import Form from './Form'
import TranslationsList from './TranslationsList'
import { fetchJobs, checkStatusAsync } from './apiCalls'

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      translations: {}
    }
    this.checkStatus = this.checkStatus.bind(this);
  }
  
  async componentDidMount() {

    try {
      const response = await fetchJobs()
      console.log('used app.js')
      if (response.jobs && response.jobs instanceof Array) {
        response.jobs.forEach((job) => {
          this.addTranslation({
            id: String(job.id),
            originalText: job.original_text,
            translatedText: job.translated_text,
            sourceLang: job.source_lang,
            targetLang: job.target_lang,
            status: job.status,
            uid: job.uid,
            timeStamp: job.timestamp * 1000
          })
        })
      }
    } catch(err) {
      this.setState({errorStatus: err.message})
    }
  }
  
  async checkStatus (itemsToCheck) {

    let response

    itemsToCheck.map(async (item) => {

      this.updateTranslation(item.id, {disabled: true})

      try {
        response = await checkStatusAsync(item)

        response && this.updateTranslation(item.id, {disabled: false})

        if (response.status === 'completed') {
          this.updateTranslation(item.id, {
            status: 'completed',
            translatedText: response.translatedText
          })
        }
      } catch (err) {
        this.setState({errorStatus: err.message})
      }
    })
  }
  
  addTranslation = ({ jobId, id, originalText, translatedText, sourceLang, targetLang, status, uid, timeStamp }) => {
    this.setState((state) => {
      return {
        translations: {
          ...state.translations,
          [id]: {
            id,
            jobId,
            originalText,
            translatedText,
            sourceLang,
            targetLang,
            status,
            uid,
            timeStamp
          }
        }
      }
    })
  }
  
  deleteTranslation = ({ id }) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          // debugger
          return translation === id ?
            newObj : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  expandTranslation = (key) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          // debugger
          return translation === key ? {
            ...newObj,
            [translation]: {
              ...state.translations[translation],
              expanded: !state.translations[translation].expanded
            }
          } : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  addUid = (key, uid) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          return translation === key ? {
            ...newObj, 
            [translation]: {
              ...state.translations[translation],
              uid,
              status: 'pending'
            }
          } : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  updateTranslation = (key, updateObj) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          return translation === key ? {
            ...newObj,
            [translation]: {
              ...state.translations[translation],
              ...updateObj
            }
          } : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  render() {
    
    const { translations } = this.state
    
    return (
      <div className='app flex-column'>
        <Label className="title"><img src={require('./utils/unbabel-logo.svg')}/>Translation Service</Label>
        <Form
          addUid={this.addUid}
          addTranslation={this.addTranslation}
          updateTranslation={this.updateTranslation}
          checkStatus={this.checkStatus}
          translations={translations}
        />
        <TranslationsList
          translations={translations}
          expandTranslation={this.expandTranslation}
          checkStatus={this.checkStatus}
          deleteTranslation={this.deleteTranslation}
        />
      </div>
    );
  }
}

export default App;
