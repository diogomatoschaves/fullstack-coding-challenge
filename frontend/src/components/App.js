import React, { Component } from 'react';
import '../App.css';
import { Label } from 'react-bootstrap'
import Form from './Form'
import TranslationsList from './TranslationsList'

class App extends Component {

  state = {
  }

  componentDidMount() {
    const headers = new Headers()

    const url = `${process.env.NODE_ENV === 'development' ? process.env.REACT_APP_API_URL_DEVELOPMENT 
      : process.env.REACT_APP_API_URL_PRODUCTION}/get_jobs`;

    fetch(url, {
      method: 'get',
      headers,
    })
      .then(res => res.json())
      .then(response => {
        if (response.jobs && response.jobs instanceof Array) {
          response.jobs.forEach((job) => {
            this.addTranslation({ 
              id: String(job.id),
              originalText: job.original_text, 
              translatedText: job.translated_text,
              sourceLang: job.source_lang,
              targetLang: job.target_lang, 
              status: job.status,
              uid: job.uid,
              timeStamp: job.timestamp * 1000
            })
          })
        }
      })
  }
  
  checkStatus = (itemsToCheck) => {
    
    const headers = new Headers()

    itemsToCheck.forEach((item) => {
      
      this.updateTranslation(item.id, {disabled: true})
      
      const url = `${process.env.NODE_ENV === 'development' ? process.env.REACT_APP_API_URL_DEVELOPMENT 
      : process.env.REACT_APP_API_URL_PRODUCTION}/check_status?uid=${item.uid}&id=${item.id}`;

      fetch(url, {
        method: 'get',
        headers,
      })
        .then(res => res.json())
        .then(response => {
          
          this.updateTranslation(item.id, {disabled: false})
          
          if (response.status === 'completed') {
            this.updateTranslation(item.id, {
              status: 'completed',
              translatedText: response.translatedText
            })
          }
        })
    })
  }
  
  addTranslation = ({ jobId, id, originalText, translatedText, sourceLang, targetLang, status, uid, timeStamp }) => {
    this.setState((state) => {
      return {
        translations: {
          ...state.translations,
          [id]: {
            id,
            jobId,
            originalText,
            translatedText,
            sourceLang,
            targetLang,
            status,
            uid,
            timeStamp
          }
        }
      }
    })
  }
  
  deleteTranslation = ({ id }) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          // debugger
          return translation === id ?
            newObj : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  expandTranslation = (key) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          // debugger
          return translation === key ? {
            ...newObj,
            [translation]: {
              ...state.translations[translation],
              expanded: !state.translations[translation].expanded
            }
          } : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  addUid = (key, uid) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          return translation === key ? {
            ...newObj, 
            [translation]: {
              ...state.translations[translation],
              uid,
              status: 'pending'
            }
          } : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  updateTranslation = (key, updateObj) => {
    this.setState((state) => {
      return {
        translations: Object.keys(state.translations).reduce((newObj, translation) => {
          return translation === key ? {
            ...newObj,
            [translation]: {
              ...state.translations[translation],
              ...updateObj
            }
          } : {
            ...newObj,
            [translation]: state.translations[translation]
          }
        }, {})
      }
    })
  }

  render() {
    
    const { translations } = this.state
    
    return (
      <div className='app flex-column'>
        <Label className="title"><img src={require('../utils/unbabel-logo.svg')}/>Translation Service</Label>
        <Form
          addUid={this.addUid}
          addTranslation={this.addTranslation}
          updateTranslation={this.updateTranslation}
          checkStatus={this.checkStatus}
          translations={translations}
        />
        <TranslationsList
          translations={translations}
          expandTranslation={this.expandTranslation}
          checkStatus={this.checkStatus}
          deleteTranslation={this.deleteTranslation}
        />
      </div>
    );
  }
}

export default App;
