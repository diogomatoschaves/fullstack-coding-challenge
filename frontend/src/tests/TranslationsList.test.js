/**
 * Created by diogomatoschaves on 26/11/2018.
 */

import React from 'react';
// import ReactDOM from 'react-dom';
import App from '../App';
import Form from '../Form';
import TranslationsList from '../TranslationsList';
import { shallow } from 'enzyme'


describe(`adding a translation`, () => {
  
  const mockTranslations = {
    "10": {
      "id": "10", 
      "jobId": undefined, 
      "originalText": "The Cross-Origin Resource Sharing standard works by adding new HTTP headers that allow servers to ",
      "sourceLang": "en", 
      "status": "pending", 
      "targetLang": "es", 
      "timeStamp": 1543174176000, 
      "translatedText": "", 
      "uid": "9e48eec7c4"}, 
    "12": {
      "id": "12", 
      "jobId": undefined, 
      "originalText": "Powered by AI and refined by a global community of tens of thousands of human linguists,", 
      "sourceLang": "en", 
      "status": "completed",
      "targetLang": "es", 
      "timeStamp": 1543174282000,
      "translatedText": "Desarrollado por AI y refinado por una comunidad global de decenas de miles de lingÃ¼istas humanos", 
      "uid": "913c966163"
    },
  }

  it('renders without crashing', () => {
    expect(shallow(<TranslationsList translations={mockTranslations} />)).toMatchSnapshot()
  });
  
  
  it('', () => {

  })

})

// describe(`adding a translation`, () => {
//
//   const mockTranslation = {
//     id: "13",
//     jobId: "dd483a4a-36d1-413d-b088-5a21cd49a589",
//     originalText: "Hi there",
//     sourceLang: {
//       key: "en",
//       name: "English"
//     },
//     status: "requesting",
//     targetLang: {
//       key: "es",
//       name: "Spanish"
//     },
//     timeStamp: 1543254667796,
//     translatedText: ""
//   }
//
//   const wrapperApp = shallow(<App />);
//   const wrapperTranlationsList = shallow(<TranslationsList />);
//   const instance = wrapperApp.instance();
//   expect(wrapperTranlationsList.props().translations).toEqual({});
//   instance.addTranslation(mockTranslation);
//   expect(wrapperTranlationsList.props().translations).toEqual({
//     13: {
//       id: "13",
//       jobId: "dd483a4a-36d1-413d-b088-5a21cd49a589",
//       originalText: "Hi there",
//       sourceLang: {
//         key: "en",
//         name: "English"
//       },
//       status: "requesting",
//       targetLang: {
//         key: "es",
//         name: "Spanish"
//       },
//       timeStamp: 1543254667796,
//       translatedText: ""
//     }
//   });


// });