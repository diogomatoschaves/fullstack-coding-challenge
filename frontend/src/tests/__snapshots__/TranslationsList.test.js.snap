// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adding a translation renders without crashing 1`] = `
<div
  className="flex-row"
  style={
    Object {
      "justifyContent": "space-around",
      "maxWidth": "850px",
      "width": "80%",
    }
  }
>
  <ListGroup
    bsClass="list-group"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="flex-row translation-item-wrapper"
      key="10"
      style={
        Object {
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <ListGroupItem
        bsClass="list-group-item"
        bsStyle="info"
        className="flex-column translation-item"
        listItem={false}
        style={
          Object {
            "border": undefined,
            "cursor": "pointer",
            "padding": 0,
            "width": "77%",
          }
        }
      >
        <div
          className="flex-column"
          onClick={[Function]}
          style={
            Object {
              "padding": "20px",
              "width": "100%",
            }
          }
        >
          <div
            className="flex-row"
            style={
              Object {
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <div
              className="message-header"
              style={
                Object {
                  "width": "40%",
                }
              }
            >
              The Cross-Origin Resource Sharing standard works by adding new HTTP headers that allow servers to 
            </div>
            <div
              style={
                Object {
                  "textAlign": "center",
                  "width": "25%",
                }
              }
            >
              <strong>
                time: 
              </strong>
              2 days ago
            </div>
            <div
              style={
                Object {
                  "textAlign": "right",
                  "width": "30%",
                }
              }
            >
              <strong>
                status: 
              </strong>
              pending
            </div>
          </div>
        </div>
      </ListGroupItem>
      <div
        style={
          Object {
            "alignSelf": "flex-start",
            "paddingTop": "20px",
            "textAlign": "center",
            "width": "20%",
          }
        }
      >
        <ButtonGroup
          block={false}
          bsClass="btn-group"
          justified={false}
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }
          vertical={false}
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="info"
            disabled={false}
            onClick={[Function]}
          >
            Update
          </Button>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="danger"
            disabled={false}
            onClick={[Function]}
          >
            Delete
          </Button>
        </ButtonGroup>
      </div>
    </div>
    <div
      className="flex-row translation-item-wrapper"
      key="12"
      style={
        Object {
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <ListGroupItem
        bsClass="list-group-item"
        bsStyle="success"
        className="flex-column translation-item"
        listItem={false}
        style={
          Object {
            "border": undefined,
            "cursor": "pointer",
            "padding": 0,
            "width": "77%",
          }
        }
      >
        <div
          className="flex-column"
          onClick={[Function]}
          style={
            Object {
              "padding": "20px",
              "width": "100%",
            }
          }
        >
          <div
            className="flex-row"
            style={
              Object {
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <div
              className="message-header"
              style={
                Object {
                  "width": "40%",
                }
              }
            >
              Powered by AI and refined by a global community of tens of thousands of human linguists,
            </div>
            <div
              style={
                Object {
                  "textAlign": "center",
                  "width": "25%",
                }
              }
            >
              <strong>
                time: 
              </strong>
              2 days ago
            </div>
            <div
              style={
                Object {
                  "textAlign": "right",
                  "width": "30%",
                }
              }
            >
              <strong>
                status: 
              </strong>
              completed
            </div>
          </div>
        </div>
      </ListGroupItem>
      <div
        style={
          Object {
            "alignSelf": "flex-start",
            "paddingTop": "20px",
            "textAlign": "center",
            "width": "20%",
          }
        }
      >
        <ButtonGroup
          block={false}
          bsClass="btn-group"
          justified={false}
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }
          vertical={false}
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="info"
            disabled={true}
            onClick={[Function]}
          >
            Update
          </Button>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="danger"
            disabled={false}
            onClick={[Function]}
          >
            Delete
          </Button>
        </ButtonGroup>
      </div>
    </div>
  </ListGroup>
</div>
`;
